library/math/conversions.lisp

```diff
@@ -0,0 +1,566 @@
+;;;; conversions.lisp
+;;;;
+;;;; Conversions between primitive numerical types
+
+(coalton-library/utils:defstdlib-package #:coalton-library/math/conversions
+  (:use
+   #:coalton
+   #:coalton-library/builtin
+   #:coalton-library/classes
+   #:coalton-library/functions
+   #:coalton-library/math/bounded)
+  (:export
+   #:integer->double-float
+   #:integer->single-float))
+
+#+coalton-release
+(cl:declaim #.coalton-impl:*coalton-optimize-library*)
+
+(in-package #:coalton-library/math/conversions)
+
+(coalton-toplevel
+  ;;
+  ;; Conversions from U8
+  ;;
+
+  (define-instance (TryInto U8 I8)
```

Comment:
I think you'd have 4-5x fewer lines with

    (define-upconversion from-small to-big)

and

    (define-downconversion from-big to-small)

macros, or something.

Reply:
Yeah probably. It's a little more annoying because all of the signed -> unsigned conversions aren't really upcasting or downcasting. Also I'd rather read the expanded code.